Client/Frontend/Browser/Search/InitialSearchEngines.swift

```diff
@@ -119,16 +134,20 @@ class InitialSearchEngines {
         if Self.yandexDefaultRegions.contains(region) {
             defaultSearchEngine = .yandex
         }
-    }
-    
-    private func priorityOverrides() {
-        guard let region = locale.regionCode else { return }
+        
+        if Self.ecosiaDefaultRegions.contains(region) {
+            replaceOrInsert(engineId: .ecosia, customId: nil)
+        }
         
         if Self.yahooEligibleRegions.contains(region) {
-            priorityEngine = .yahoo
+            replaceOrInsert(engineId: .yahoo, customId: nil)
         }
     }
     
+    private func priorityOverrides() {
+        // No priority engines are live at the moment.
```

Comment:
Asking it for context, when there is no other, our default priority engine is google in this case isnt it ?

Reply:
Priority engine is a special case, it is not set as default, but on onboarding it is shown at the top of the list.
For user added engines you work on there is no priority-engine logic, it's for predefined engines only

So full list goes like

1. Priority engine
2. Default engine(preselected)
3. other engine
4. other engine
..